libtiledbsoma/src/reindexer/reindexer.cc

```diff
@@ -0,0 +1,202 @@
+/**
+ * @file   reindexer.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2022 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file defines the a pybind11 api into SOMA C++ library.
+ */
+
+#include "reindexer.h"
+#include <unistd.h>
+#include "khash.h"
+
+uint64_t current_time() {
+    uint64_t microseconds_since_epoch =
+        std::chrono::duration_cast<std::chrono::microseconds>(
+            std::chrono::system_clock::now().time_since_epoch())
+            .count();
+    return microseconds_since_epoch;
+}
+
+static std::mutex log_mutex;
+
+static bool no_logging_s = true;
+
+static void log(std::string message, int ln, std::string severity = "INFO") {
```

Comment:
Please replace log usage with the existing logger, https://github.com/single-cell-data/TileDB-SOMA/blob/main/libtiledbsoma/src/utils/logger.h.

I.e: https://github.com/single-cell-data/TileDB-SOMA/blob/af2ead33888de1405be5cec0c91784b78880eec4/libtiledbsoma/src/soma/managed_query.cc#L116